frontend/src/events/EventCard.tsx

```diff
@@ -1,26 +1,66 @@
-import { Card, Button } from "react-bootstrap";
+import { useState } from "react";
+import { Container, Card, Button } from "react-bootstrap";
 
 import { Event } from "../interfaces/Event";
+import { UpdateEventFormData } from "../interfaces/UpdateEventFormData";
+
+import { UpdateEventForm } from "./UpdateEventForm";
 
 interface EventCardParamsInterface {
   event: Event;
-  remove: (eventId:number) => void
+  updateEvent: (event: Event, formData: UpdateEventFormData) => void;
+  removeEvent: (eventId: number) => void;
+  toggleDynamicMarker: (show: boolean) => void;
+  isAddingEvent: boolean;
 }
 
-export function EventCard({ event, remove }: EventCardParamsInterface) {
+export function EventCard({
+  event,
+  updateEvent,
+  removeEvent,
+  toggleDynamicMarker,
+  isAddingEvent,
+}: EventCardParamsInterface) {
+  const [isUpdatingEvent, setIsUpdatingEvent] = useState(false);
+
   const { title, type, id } = event;
 
+  //Toggle Add Event Button
+  function toggleUpdateEvent() {
+    setIsUpdatingEvent(true);
+    toggleDynamicMarker(true);
+  }
+
+  async function handleCancel() {
+    setIsUpdatingEvent(false);
+    toggleDynamicMarker(false);
+  }
+
   async function handleRemove() {
-    await remove(id);
+    toggleDynamicMarker(false);
+    await removeEvent(id);
   }
 
   return (
     <Card className="mb-3">
-      <Card.Body>
-        <Card.Title>{title}</Card.Title>
-        <Card.Text>{type}</Card.Text>
-        <Button onClick={handleRemove}>Remove</Button>
-      </Card.Body>
+      {isUpdatingEvent ? (
+        <Card.Body>
+          <UpdateEventForm event={event} updateEvent={updateEvent} />
+          <Button onClick={handleCancel}>Cancel</Button>
+        </Card.Body>
+      ) : (
+        <Card.Body>
+          {" "}
+          <Card.Title>{title}</Card.Title>
+          <Card.Text>{type}</Card.Text>
+          {!isAddingEvent ? (
+            <Container>
+              <Button onClick={toggleUpdateEvent}>Update</Button>
+              <Button onClick={handleRemove}>Remove</Button>
+            </Container>
+          ) : null}
```

Comment:
A user can update one event, and then start editing another, revealing both events in an update state. May be fixed when we refactor to move adding events to a separate page, but just noting this for future reference.